src/emu/image.cpp

```diff
@@ -17,6 +19,13 @@
 #include "xmlfile.h"
 
 //**************************************************************************
+//  STATIC VARIABLES
+//**************************************************************************
+
+static std::regex s_softlist_regex("\\w+\\:\\w+\\:\\w+");
```

Comment:
well, trying to be consistent with MAME tags and shortnames, we have always
tried to stick to letters/digits/underscores only as valid listname +
shortname + part, so that \w seems ok in this occurrence
in particular, it makes life much easier when launching from command line
across most OS and keyboards ;)

   f.


On Mon, Aug 1, 2016 at 4:36 AM, Vas Crabb <notifications@github.com> wrote:

> In src/emu/image.cpp
> <https://github.com/mamedev/mame/pull/1161#discussion_r72918455>:
>
> > @@ -17,6 +19,13 @@
> >  #include "xmlfile.h"
> >
> >  //**************************************************************************
> > +//  STATIC VARIABLES
> > +//**************************************************************************
> > +
> > +static std::regex s_softlist_regex("\\w+\\:\\w+\\:\\w+");
>
> wouldn't it be better to use a string like "[^:/]+\:[^:/]+\:[^:/]+"
> working from unacceptable characters rather than a narrow class to include?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mamedev/mame/pull/1161/files/024f698364b2d7396fe800e76cc0e10d1a19da6d#r72918455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI0qvIHYYCUhcneQhgWPkdwJL_E6h1Biks5qbVuqgaJpZM4JY0u->
> .
>